### ISSUE TITLE
Flashing error boundary when navigating to route with loader function in Firefox

### ISSUE THREAD

**Issue by haakonmt** [sergiodxa/remix-i18next#29]
After setting up `remix-i18next`, I'm seeing a flashing error boundary when I navigate to a route which defines a loader function. 

I am able to consistently reproduce the issue in Firefox (developer edition 96.0b10), but not in Chrome, neither on desktop or mobile.

I'm not quite sure what the issue is, but the route module seems to be undefined at some point it shouldn't be. I've tried to make a minimal repro of the bug:
- https://github.com/haakonmt/remix-i18next-bug-repro
- https://remix-i18next-bug-repro.vercel.app/

![gif of reproduction](https://s10.gifyu.com/images/repro.gif)

**Comment by danestves** [comment]
Also integrating that inside the `entry.client.tsx` makes the remix app refresh after click any internal URL

**Comment by SaeedSheikhi** [comment]
Same issue, Application Error on Firefox and Client-side routing doesn't work correctly, cause whole page refresh

**Comment by sergiodxa** [comment]
After trying to debug this, I found out if you copy the code of https://github.com/sergiodxa/remix-i18next/blob/35547edb796c3fc3739f7d74c544d5a2b4cfb8bb/src/react.tsx#L1-L50 to your app code, everything will work without issues.

I'm still trying to debug what's happening, anyone with this error can do that to fix it.

**Comment by willin** [comment]
i did not see any errors...  but index page refreshing again and again

- Errored: https://init.willin-wang.pages.dev/
- Normal:  https://init.willin-wang.pages.dev/en

<https://github.com/willin/willin.wang/tree/init>

**Comment by matmilbury** [comment]
I experience too. It seems to be related to https://github.com/remix-run/remix/issues/1618

Most likely, the server code gets bundled with the browser code.

**Comment by vmorenog** [comment]
Hello! For those who are facing the same issue with the reload of the pages, until there's a fix, the one mentioned by @sergiodxa will work. Just create a new file in your codebase e.g. `src/app/i18/Remixi18NextProvider.tsx` copy the whole file referenced https://github.com/sergiodxa/remix-i18next/issues/29#issuecomment-1013815861 and, if needed, paste in the same file the types for `Language`:

```
type Value = string | Language;
type Key = string;
export type Language = { [key: Key]: Value };
```

Then, change all the imports of `<RemixI18NextProvider>` to point to your file. 

So if you had: `import { RemixI18NextProvider } from 'remix-i18next'`
Now it will be: `import { RemixI18NextProvider } from '~/i18n/RemixI18NextProvider'`

Hope it helps!

**Comment by sergiodxa** [comment]
This is fixed with v4.0.0. It's not a huge change but it change many things of how the library works and integrates with i18next ecosystem.
